### PR TITLE
command to restart the REPL without killing the buffer

### PR DESCRIPTION
**Pull request by boehm-s**
A while ago, I answered [this question](https://stackoverflow.com/questions/43465133/repl-driven-development-in-node-js/43596508#43596508), and I think the command could be useful to other users.

I'm not experimented at all in elisp, so I'm open to all kind of suggestions,

Cheers

### CODE REVIEW

**Comment by abicky** [comment]
Thank you for your PR!
I think it is better to introduce `nodejs-repl-use-global` custom variable so that `.clear` command clears the current context. The current implementation sets `useGlobal` to true as node CLI REPL does.
cf. https://nodejs.org/api/repl.html#repl_repl_start_options

What do you think?

```
% node -e "require('repl').start({ replMode: 'magic', useGlobal: true })"
> let foo = 1;
undefined
> .clear
> let foo = 2;
TypeError: Identifier 'foo' has already been declared
> 
% node -e "require('repl').start({ replMode: 'magic', useGlobal: false })"
> let foo = 1;
undefined
> .clear
Clearing context...
> let foo = 2;
undefined
```

**Comment by boehm-s** [comment]
I haven't tought about that, it's way better, and cleaner for sure !

**Comment by boehm-s** [comment]
Do you think it's OK like that ?

File: nodejs-repl.el [hunk 1]
```diff
@@ -135,7 +139,7 @@ See also `comint-process-echoes'"
 
 (defvar nodejs-repl-code-format
   (concat
-   "require('repl').start('%s', null, null, true, false, "
+   "require('repl').start({prompt: '%s', useGlobal: %s}, null, null, true, false, "
```
Comment (abicky) [inline nodejs-repl.el]: `repl.start` has two signatures, one is `repl.start(options)` and another is `repl.start(prompt, stream, eval, useGlobal, ignoreUndefined, replMode)`. If you specify a object as the first argument, other arguments are ignored, so please change the code like below:

```diff
diff --git a/nodejs-repl.el b/nodejs-repl.el
index 58834bb..32f758c 100644
--- a/nodejs-repl.el
+++ b/nodejs-repl.el
@@ -139,8 +139,8 @@ See also `comint-process-echoes'"

 (defvar nodejs-repl-code-format
   (concat
-   "require('repl').start({prompt: '%s', useGlobal: %s}, null, null, true, false, "
-   "require('repl')['REPL_MODE_' + '%s'.toUpperCase()])"))
+   "require('repl').start({ prompt: '%s', useGlobal: %s, replMode: "
+   "require('repl')['REPL_MODE_' + '%s'.toUpperCase()] })"))

 (defvar nodejs-repl-extra-espace-sequence-re "\\(\x1b\\[[0-9]+[GJK]\\)")

```

**Comment by abicky** [comment]
I've published 0.2.1.

**Pull request merged**